lib/client.js

```diff
@@ -531,47 +538,15 @@ function Request(method, url) {
 }
 
 /**
- * Mixin `Emitter`.
+ * Mixin `Emitter` and `RequestBase`.
  */
 
 Emitter(Request.prototype);
```

Comment:
Oh, I didn't realize the Stream complication.

What confused me was that `RequestBase` is called in constructor, which looks like "single inheritance" pattern to me. I'd expect JS classes to either emulate single inheritance, or patch prototypes at will, but not both.

We can't fully use single inheritance pattern (with full prototype chain, Object.create, etc.), so in that case I'd go all the way the other way and make it clear RequestBase isn't a class, but only a bunch of functions to copy.

So my suggestion is to remove call to RequestBase in constructor. Perhaps even instead of using function.prototype, just assign functions to plain object `module.exports = {method: function(){}}`, so it's clear it's not a base class.